### PR TITLE
Unmerged features

### PR DESCRIPTION
**Pull request by muhamadSammar**

### CODE REVIEW

File: dr_trigger.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#! /usr/bin/env python3
+
+import socket
+import sys
+import os
+import io
+import struct
+
+def open_server(server_pid):
```
Comment (alexvesker) [inline dr_trigger.py]: connect_to_server

File: dr_trigger.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    path = "/var/tmp/dpdk_net_mlx5_%d" % server_pid
```
Comment (alexvesker) [inline dr_trigger.py]: should not be hardcoded

File: dr_trigger.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
```
Comment (alexvesker) [inline dr_trigger.py]: check if file exists

File: dr_trigger.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+def request_dump(sock, port, fd):
+    sock.sendmsg([struct.pack('H', port)],
+                 [(socket.SOL_SOCKET, socket.SCM_RIGHTS, struct.pack('i',fd))])
+    msg,ancdata,flags,addr = sock.recvmsg(struct.calcsize('i'))
+    ret = struct.unpack('i', msg)[0]
+    if ret:
+        print("server error: %s" % os.strerror(ret))
+
+def usage():
```
Comment (alexvesker) [inline dr_trigger.py]: is it needed?

File: mlx_steering_dump.py [hunk 5]
```diff
@@ -280,11 +281,18 @@ def main():
...
-		return 0
+	file_path = ""
+	if (args.FILEPATH != ""):
+		file_path = args.FILEPATH
+	elif (len(args.trigger) == 2):
+		pid = int(args.trigger[0])
+		port = int(args.trigger[1])
+		file_path = dr_trigger.trigger_dump(pid, port)
+	else:
+		print_dr("Neither input file or trigger option provided (-f FILEPATH or --trigget <pid> <port>)")
```
Comment (alexvesker) [inline mlx_steering_dump.py]: -triggrt -> -d <path to file>

Comment (muhamadSammar) [reply mlx_steering_dump.py]: done

File: dr_trigger.py (hunk 4 above)
Comment (muhamadSammar) [reply dr_trigger.py]: no, deleted

File: dr_trigger.py (hunk 2 above)
Comment (muhamadSammar) [reply dr_trigger.py]: done

File: dr_trigger.py (hunk 3 above)
Comment (muhamadSammar) [reply dr_trigger.py]: no need, exception will be caught

File: mlx_steering_dump.py [hunk 6]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python
```
Comment (alexvesker) [inline mlx_steering_dump.py]: #!/usr/bin/env python3

File: mlx_steering_dump.py [hunk 7]
```diff
@@ -269,6 +269,7 @@ def parse_args():
 	parser = argparse.ArgumentParser(
 	description='''mlx_steering_dump.py - Steering dump tool''')
 	parser.add_argument('-f', dest="FILEPATH", default="", help='input steering dump file path')
+	parser.add_argument('-d', dest="trigger", metavar=('pid', 'port', 'path'), nargs=3, help='<server pid> <port number> <FILE PATH>')
```
Comment (alexvesker) [inline mlx_steering_dump.py]: <SERVER PID> <PORT NUMBER>

Comment (alexvesker) [reply mlx_steering_dump.py]: <SERVER PID> <PORT NUMBER>

File: mlx_steering_dump.py [hunk 8]
```diff
@@ -284,6 +285,14 @@ def main():
...
 		return 0
 
+	if (len(args.trigger) == 2):
+		pid = int(args.trigger[0])
+		port = int(args.trigger[1])
+		dr_trigger.trigger_dump(pid, port, args.FILEPATH)
+	elif(os.path.exists(args.FILEPATH) == False):
+		print_dr("Input steering dump file doesn't exist")
+		return 0
+
```
Comment (alexvesker) [inline mlx_steering_dump.py]: remove

**Pull request merged**